### PR TITLE
Complete generating data and seeding for couchdb

### PR DESCRIPTION
**Pull request by maxmilpro**
Installed and setup couchdb node. Generated 10M records and inserted into couchdb `images` database

- reused generateCSV() to write 10M records to couch.csv - documents in couchdb required different schema and structure than mysql
- added npm script to bulk insert csv into couchdb using [couchimport npm libray](https://www.npmjs.com/package/couchimport)
- cleaned up a few missing semicolons :)

### CODE REVIEW

File: seed/Page.js [hunk 1]
```diff
@@ -12,3 +12,14 @@ module.exports.createMainImageRecords = (id) => {
...
+
+module.exports.createCouchRecord = (id) => {
+  const randomPage = randomPageSelector();
+  const mainImages = [...new Array(8)].map((currentelement, index) => {
+    return JSON.stringify({
+      main: `https://steam-fec.s3.amazonaws.com/steam${randomPage}/main-${randomPage}-${index}.jpg`,
+      thumb: `https://steam-fec.s3.amazonaws.com/steam${randomPage}/thumb-${randomPage}-${index}.jpg`
+    })
+  })
+  return `${id}\thttps://steam-fec.s3.amazonaws.com/steam${randomPage}/header-${randomPage}.jpg\t[${mainImages}]\n`
```
Comment (jamesh48) [inline seed/Page.js:24]: I want to know what \t does, I know that was a big discovery for you from the slack chat

File: seed/seedingScript.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
 const fs = require('fs');
 const path = require('path');
 const { generateCSV, generateJSON } = require('./generateData');
-const { createHeaderImageRecord, createMainImageRecords } = require('./Page.js');
+const { createHeaderImageRecord, createMainImageRecords, createCouchRecord } = require('./Page.js');
 const filePath = path.join(__dirname, '/data');
 
 try {
   const headerImagesStream = fs.createWriteStream(filePath + '/headerImages.csv');
   generateCSV(headerImagesStream, createHeaderImageRecord, 1e7, () => {
```
Comment (jamesh48) [inline seed/seedingScript.js:9]: this sounds like a handy package, the function takes in the stream, a data generation function, presumably a number of times to write and a callback, right?

File: seed/seedingScript.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-    headerImagesStream.end()
-    console.log('Successfully wrote headerImage records to CSV')
+    headerImagesStream.end();
+    console.log('Successfully wrote headerImage records to CSV');
   });
 } catch (err) {
```
Comment (jamesh48) [inline seed/seedingScript.js:13]: catch (err) 
  if (err) ... 

I think you can just log that err, if its caught it's an error i'm pretty sure, i'm sure this was just an oversight though

File: seed/seedingScript.js [hunk 4]
```diff
@@ -18,8 +18,19 @@ try {
   const mainImagesStream = fs.createWriteStream(filePath + '/mainImages.csv');
   generateCSV(mainImagesStream, createMainImageRecords, 1e7, () => {
     mainImagesStream.end();
-    console.log('Successfully wrote mainImage records to CSV')
+    console.log('Successfully wrote mainImage records to CSV');
   });
 } catch (err) {
   if (err) {console.log(err)}
 }
```
Comment (jamesh48) [inline seed/seedingScript.js:25]: well here it is again, are you throwing something else somewhere? I think the err first if statement only applies with call backs, in the try/catch scheme its an error in the catch block, right?

File: seed/seedingScript.js [hunk 5]
```diff
@@ -18,8 +18,19 @@ try {
...
+
+try {
+  const couchStream = fs.createWriteStream(filePath + '/couch.csv');
+  couchStream.write('_id\theaderImage\tmainImages\n');
```
Comment (jamesh48) [inline seed/seedingScript.js:29]: oh ok so is the \t a short hand way of separating csv values? i've had success with just using a comma, but could begin to imagine how the \t might be more dynamic and interesting

File: seed/seedingScript.js [hunk 6]
```diff
@@ -18,8 +18,19 @@ try {
...
+  generateCSV(couchStream, createCouchRecord, 1e7, () => {
+    couchStream.end();
+    console.log('Successfully wrote couch records to CSV');
+  })
+} catch (err) {
+  if (err) {console.log(err)}
+}
```
Comment (jamesh48) [inline seed/seedingScript.js:36]: Looks good overall!

**Review by jamesh48 — APPROVED** [review]
Nice job! I'm not quite sure if this is the whole script or if there are other parts, but this seems pretty minimal and understandable. I like the csv generator npm package you are using, seems straightforward..maybe even optimized? Still just scratching my head about how people come up with some of the db names.

File: seed/Page.js (hunk 1 above)
Comment (maxmilpro) [reply seed/Page.js:24]: `\t` is a tab. i was trying to insert a string with commas into mysql but i was running into a problem. the default column delimeter/separator for `LOAD DATA LOCAL` is a comma `,` so it thought every comma was a new column/value. 

`'value','some,string,with,commas'` => 5 columns

i was able to set the delimeter/separator to `\t` and separate my columns with tabs.

`'value'    'some,string,with,commas'` => 2 columns

File: seed/seedingScript.js (hunk 2 above)
Comment (maxmilpro) [reply seed/seedingScript.js:9]: i wrote this function myself. you can find it in `seed/generateData.js`. and yes, those are exactly the parameters 😄  i'm pretty proud if this one!

```
const fs = require('fs');
const path = require('path');

const timer = (record, start) => {
  if (record % 100000 === 0) {
    const now = Date.now();
    const minutesElapsed = (((now - start) / 1000) / 60);
    console.log(`created record #${record} - minutes elapsed: ${minutesElapsed}`);
  }
}

const start = Date.now();

module.exports.generateCSV = (writer, data, records, cb) => {
  let i = 0;
  write();
  function write() {
    let ok = true;
    do {
      i++;
      if (i === records) {
        writer.write(data(i), cb);
      } else {
        ok = writer.write(data(i));
        timer(i, start);
      }
    } while (i < records && ok);
    if (i < records) {
      writer.once('drain', write);
    }
  }
};
```

File: seed/seedingScript.js (hunk 4 above)
Comment (maxmilpro) [reply seed/seedingScript.js:25]: yeah i think youre right about this one. i definitely just threw it in there to have something in the catch block. good call! i will update

File: seed/seedingScript.js (hunk 5 above)
Comment (maxmilpro) [reply seed/seedingScript.js:29]: yup, exactly. you can separate your csv with whatever value you want, ie commas, tabs, spaces, etc. technically you could separate it with a string like `'separator'` and then you have to tell whatever import tool how youre separating columns.

**Pull request merged**